### PR TITLE
fix: AvaloniaUI/Avalonia#12974 - Preview not showed when using using statment

### PR DESCRIPTION
**Pull request by workgroupengineering**
## Current behavior

SreTypeSystem find only public types

## Expected behavior

SreTypeSystem find only public and internal types

part of https://github.com/AvaloniaUI/Avalonia/issues/12974

### CODE REVIEW

File: src/XamlX/Extensions/TypeExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,46 @@
+﻿namespace System;
+
+/// <summary>
+/// Source https://stackoverflow.com/a/46382137/20894223
+/// </summary>
+internal static class TypeExtensions
+{
+    public static bool IsInternal(this Type t) => !t.IsVisible
```
Comment (kekekeks) [inline src/XamlX/Extensions/TypeExtensions.cs]: The method explicitly excludes nested internal types, so the naming is clearly wrong here.

File: src/XamlX/Extensions/TypeExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+            && !t.IsNested
+            && !t.IsNestedPublic
+            && !t.IsNestedFamily
+            && !t.IsNestedPrivate
+            && !t.IsNestedAssembly
+            && !t.IsNestedFamORAssem
+            && !t.IsNestedFamANDAssem;
+
+    // only nested types can be declared "protected"
+    public static bool IsProtected(this Type t) => !t.IsVisible
```
Comment (kekekeks) [inline src/XamlX/Extensions/TypeExtensions.cs]: Do we need those two methods?

Comment (workgroupengineering) [reply src/XamlX/Extensions/TypeExtensions.cs]: Not at the moment, I'll remove them

**Comment by workgroupengineering** [comment]
Do I need to make other changes?

**Review by kekekeks — APPROVED** [review]

**Pull request merged**